### ISSUE TITLE
 Liveness and Readiness probes fail with TLS configuration.

### ISSUE THREAD

**Issue by nmagnezi** [observatorium/observatorium#60]
While working on [PR-273](https://github.com/observatorium/configuration/pull/273) I got a working API Pod that never gets marked as ready.

The description revealed that both  Liveness and Readiness probes fail:
```
 Warning  Unhealthy  2m52s (x3 over 3m52s)   kubelet, kind-control-plane  Liveness probe failed: HTTP probe failed with statuscode: 503
 Warning  Unhealthy  2m46s (x19 over 4m15s)  kubelet, kind-control-plane  Readiness probe failed: HTTP probe failed with statuscode: 503
```

I had to [work around it and enable TCP probes](https://github.com/observatorium/configuration/pull/273/commits/a6298d3bd1c03f5325f52f831599569f571e939c) to enable the Pod, but this is not a suitable replacement as it does not actually check the application.

The deployment manifest:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/component: api
    app.kubernetes.io/instance: observatorium-xyz
    app.kubernetes.io/name: observatorium-api
    app.kubernetes.io/part-of: observatorium
    app.kubernetes.io/version: master-2020-05-27-v0.1.1-67-gb2eb292
  name: observatorium-xyz-observatorium-api
  namespace: observatorium
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/component: api
      app.kubernetes.io/instance: observatorium-xyz
      app.kubernetes.io/name: observatorium-api
      app.kubernetes.io/part-of: observatorium
  strategy:
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 1
  template:
    metadata:
      labels:
        app.kubernetes.io/component: api
        app.kubernetes.io/instance: observatorium-xyz
        app.kubernetes.io/name: observatorium-api
        app.kubernetes.io/part-of: observatorium
        app.kubernetes.io/version: master-2020-05-27-v0.1.1-67-gb2eb292
    spec:
      containers:
      - args:
        - --web.listen=0.0.0.0:8080
        - --web.internal.listen=0.0.0.0:8081
        - --logs.read.endpoint=http://127.0.0.1
        - --logs.write.endpoint=http://127.0.0.1
        - --metrics.read.endpoint=http://observatorium-xyz-cortex-query-frontend.observatorium.svc.cluster.local:9090
        - --metrics.write.endpoint=http://observatorium-xyz-thanos-receive.observatorium.svc.cluster.local:19291
        - --log.level=warn
        - --tls-cert-file=/mnt/certs/server.pem
        - --tls-private-key-file=/mnt/certs/server.key
        - --tls-client-ca-file=/mnt/certs/ca.pem
        - --tls-reload-interval=1m
        image: quay.io/observatorium/observatorium:master-2020-05-27-v0.1.1-67-gb2eb292
        livenessProbe:
          failureThreshold: 10
          httpGet:
            path: /live
            port: 8081
            scheme: HTTP
          periodSeconds: 30
        name: observatorium-api
        ports:
        - containerPort: 8081
          name: internal
        - containerPort: 8080
          name: public
        readinessProbe:
          failureThreshold: 12
          httpGet:
            path: /ready
            port: 8081
            scheme: HTTP
          periodSeconds: 5
      volumes:
      - name: observatorium-api-tls-secret
        secret:
          secretName: observatorium-api-tls-secret
```

The issue should probably be handled in observatorium-api code.

**Comment by metalmatze** [comment]
If 503 are returned, that currently means that the upstream application isn't proxy-able by the API. For those HTTP readiness probes to succeed it needs the backend to be available. Liveness probes should still work fine, if the HTTP server is up to begin with.

**Comment by squat** [comment]
fixed in #66 :)
